### PR TITLE
add copy button

### LINKED ISSUE #1: Add a copy link option?
I think many would like to copy the link to the thread as I feel that's the main use case here. Currently we have to right click and copy link. Providing an option to copy the link would be awesome. Probably like some small copy icon. 

Or we can change the current # to act as a copy instead of a link?

Thoughts?

### PR DESCRIPTION
**Pull request by karuppiah7890**
Fixes #1 

This is still a draft because I still haven't tested the permissions for the clipboard write feature. Been pending for long now. Will do it soon! :)

### CODE REVIEW

**Comment by karuppiah7890** [comment]
Also, do comment on the code! @sriv

**Comment by karuppiah7890** [comment]
For UI, I just used `@` character. Let me know what better we can do if that doesn't look good or intuitive

File: thread-link.js [hunk 1]
```diff
@@ -7,16 +7,33 @@
...
+        if (!document.getElementById(threadLinkButtonId)) {
             var a = document.createElement('a');
             a.href = threadLink;
             a.title = "Permalink";
             a.innerText = "#";
             a.style = "padding: 0 0.3em 0 0.5em;font-weight: bold;";
             var followWrapper = follow.parentNode.parentNode;
             followWrapper.appendChild(a);
+
+            var span = document.createElement('span')
```
Comment (sriv) [inline thread-link.js]: creating another element is redundant. I think we could just change the behaviour of `a` to 

- copy url on click
- ensure `return false` to preventDefault , i.e. navigation.

File: thread-link.js [hunk 2]
```diff
@@ -7,16 +7,33 @@
...
+            span.onclick = copyToClipboardFunc(threadLink);
+            followWrapper.appendChild(span);
         }
-    }    
-})();
\ No newline at end of file
+    }
+})();
+
+function copyToClipboardFunc(link) {
```
Comment (sriv) [inline thread-link.js]: this could just be inlined. It has only one usage.

File: thread-link.js [hunk 3]
```diff
@@ -7,16 +7,33 @@
...
+    return function () {
+        navigator.clipboard.writeText(link).then(function () {
+            console.log("successfully copied text!")
```
Comment (sriv) [inline thread-link.js]: no need to log on success, I think we could just log on failures.

**Comment by sriv** [comment]
works on FF 76. Thanks for sending this patch. Some comments inline.

**Review by sriv — CHANGES_REQUESTED** [review]

File: thread-link.js (hunk 1 above)
Comment (karuppiah7890) [reply thread-link.js]: Okay! 👍

File: thread-link.js (hunk 2 above)
Comment (karuppiah7890) [reply thread-link.js]: Will do 👍

File: thread-link.js (hunk 3 above)
Comment (karuppiah7890) [reply thread-link.js]: Makes sense 👍

File: thread-link.js (hunk 1 above)
Comment (karuppiah7890) [reply thread-link.js]: Done!

**Comment by karuppiah7890** [comment]
There's just one thought I had - now the function is independent and the same for all elements - can we extract it out? So that we don't create an anonymous function for every element? That's not necessary. Too many functions

**Comment by karuppiah7890** [comment]
You mentioned inlining function. Can we extract this time? Given it's a general function, and not a closure

**Comment by sriv** [comment]
sounds good @karuppiah7890

**Comment by karuppiah7890** [comment]
@sriv It's done! 👍

**Review by sriv — APPROVED** [review]

**Pull request merged**